cypress/e2e/exchange.cy.js

```diff
@@ -0,0 +1,95 @@
+const URL = 'http://127.0.0.1:8080'
+const API_URL = 'https://api.exchangerate.host/'
+const VALID_DATE = '2014-08-15'
+const BASE = 'ARS'
+const CONVERT_TO = 'USD'
+const TODAYS_YEAR = 2023
+
+context('Exchange rate app', () => {
+  before(() => {
+    cy.visit(URL)
+  })
+
+  describe('Test the application', () => {
+    it('should show the error when quantity is not specified', () => {
+      cy.get('#from-currency').select(BASE)
+      cy.get('#to-currency').select(CONVERT_TO)
+      cy.get('#date').type(VALID_DATE)
+
+      cy.get('#submit-form').click()
+      cy.get('#quantity').should('have.class', 'bg-danger')
```

Comment:
This one I don't understand it very well, do you mean saving the "cy.get('#quantity)" as an alias for applying the should chain?
Wouldn't this end up being something like:
cy.get('#quantity').as('quantity')
cy.get('@quantity').should('have.class', 'bg-danger')
and for me, this doesn't have that much sense since you are practically rewriting the cy.get() call

Reply:
It can be saved as `@quant` but the only thing this does is saving a few letters.

Reply:
In that case, you have 2 options.
- Don't add an alias to cy.get("#quantity")
- Change quantity id

Reply:
I think I get it now, I'm looking for a more readable way to know what that "quantity" is by reading its name, right? In that case I could save it as an alias like ` cy.get("#quantity")as.("quantity-input")` so whoever reads it can get more info about what it actually is.

Reply:
Yeah but I think `quantity` wasn't a good example.

Check these two tests:

```
it('should render the correct table data after calling the API with a provided date', () => {
      cy.get('tbody tr').should('have.length.of.at.least', 5) <--- This could be an alias
      cy.get('tbody > *:first-child > *:first-child').should('have.text', 'USD') <--- This could be an alias
      cy.get('tbody > *:first-child > *:last-child').should( <--- This could be an alias
        'have.text',
        '0,121'
      )
    })
```

```
it('should render the correct table data after calling the API no provided date', () => {
      cy.get('tbody tr').should('have.length.of.at.least', 5) <--- This could be an alias
      cy.get('tbody > *:first-child > *:first-child').should('have.text', 'AED') <--- This could be an alias
      cy.get('tbody > *:first-child > *:last-child').should( <--- This could be an alias
        'have.text',
        '0,018'
      )
    })
```

What I suggest is to do something like this:

```
describe('Test the application', () => {
  before(() => {
    cy.visit(URL)
    cy.get('tbody tr').as("tBodyTr") 
    cy.get('tbody > *:first-child > *:first-child').as("firstChild")
    cy.get('tbody > *:first-child > *:last-child').as("lastChild")
  })
it('should render the correct table data after calling the API with a provided date', () => {
      cy.get('@tBodyTr').should('have.length.of.at.least', 5)
      cy.get('@firstChild').should('have.text', 'USD')
      cy.get('@lastChild').should(
        'have.text',
        '0,121'
      )
    })
    
it('should render the correct table data after calling the API no provided date', () => {
      cy.get('@tBodyTr').should('have.length.of.at.least', 5)
      cy.get('@firstChild').should('have.text', 'AED')
      cy.get('@lastChild').should(
        'have.text',
        '0,018'
      )
    })
})
```
I think is more legible and clean.

Reply:
Where are the changes? Have you committed it? Can't see it.

Reply:
I forgot to push it, it should be visible now

Reply:
Ok. Do you remember [this](https://github.com/RamunnoAJ/casa-de-cambio/pull/1#discussion_r1145338778)? 

I think the structure should be like this:
```
describe('Test the application', () => {
  before(() => {
    cy.visit(URL)
    cy.get('tbody tr').as("tBodyTr") 
    cy.get('tbody > *:first-child > *:first-child').as("firstChild")
    cy.get('tbody > *:first-child > *:last-child').as("lastChild")
  })

it('should render the correct table data after calling the API with a provided date', () => {
      cy.get('@tBodyTr').should('have.length.of.at.least', 5)
      cy.get('@firstChild').should('have.text', 'USD')
      cy.get('@lastChild').should(
        'have.text',
        '0,121'
      )
    })
    
it('should render the correct table data after calling the API no provided date', () => {
      cy.get('@tBodyTr').should('have.length.of.at.least', 5)
      cy.get('@firstChild').should('have.text', 'AED')
      cy.get('@lastChild').should(
        'have.text',
        '0,018'
      )
    })
})
```

Just one describe block containing the tests inside.

You can read more about the best practices [here](https://docs.cypress.io/guides/references/best-practices)

Reply:
The problem using only one describe block containing all of my test appears when at the start of the app I ask for a cypress to get a table that it doesn't exist yet, so the 
```before(() => {
  cy.visit(URL)
  cy.get('tbody tr').as("tBodyTr") 
  cy.get('tbody > *:first-child > *:first-child').as("firstChild")
  cy.get('tbody > *:first-child > *:last-child').as("lastChild")
})```
will fail because it can't get the table yet. A solution that occurred to me was separating the tests in two describe blocks, one for the tables test and the other for the rest, but this doesn't follow along with best practices. My code would end up with two describe blocks who are basically working on the same page but different UI components.